test/xbgpu/test_engine.py

```diff
@@ -892,3 +894,27 @@ def heap_factory(batch_index: int) -> list[spead2.send.HeapReference]:
             prom_diff.get_sample_diff("output_x_clipped_visibilities_total", {"stream": "bcp1"})
             == n_channels_per_substream * n_baselines
         )
+
+    @DEFAULT_PARAMETERS
+    async def test_bad_requests(self, client: aiokatcp.Client, n_ants: int) -> None:
+        # Trying to use beamformer request on wrong stream type
+        with pytest.raises(aiokatcp.FailReply, match=r"not a tied-array-channelised-voltage stream"):
+            await client.request("beam-quant-gains", "bcp1", 1.0)
+        with pytest.raises(aiokatcp.FailReply, match=r"not a tied-array-channelised-voltage stream"):
+            await client.request("beam-weights", "bcp1", *([1.0] * n_ants))
+        with pytest.raises(aiokatcp.FailReply, match=r"not a tied-array-channelised-voltage stream"):
+            await client.request("beam-delays", "bcp1", *(["0.0:0.0"] * n_ants))
+
+        # Vector requests with wrong number of parameters
+        with pytest.raises(aiokatcp.FailReply, match=r"Incorrect number of weights \(expected 4, received 3\)"):
+            await client.request("beam-weights", "beam_0x", 1.0, 2.0, 3.0)
+        with pytest.raises(aiokatcp.FailReply, match=r"Incorrect number of delays \(expected 4, received 5\)"):
+            await client.request("beam-delays", "beam_0x", "0:0", "1:1", "2:2", "3:3", "4:4")
+
+        # Bad delay formatting
+        with pytest.raises(aiokatcp.FailReply):
+            await client.request("beam-delays", "beam_0x", "0", "1", "2", "3")  # Missing ":"
+        with pytest.raises(aiokatcp.FailReply):
+            await client.request("beam-delays", "beam_0x", "0:0", "1:1", "2:2", "3:3:3")  # Too many :'s
+        with pytest.raises(aiokatcp.FailReply):
+            await client.request("beam-delays", "beam_0x", "0:0", "1:1", "2:2", "3:2j")  # Not float
```

Comment:
I gather the `2j` causes the failure, because the other elements/numbers/values are the same as the previous request.